pyvisa_sim/component.py

```diff
@@ -200,3 +239,44 @@ def _match_setters(self, query):
                 return self.error_response("command_error")
 
         return None
+    
+    def _match_connection(self, query, connections=None):
+        """Tries to match in connections
+
+        :param query: message tuple
+        :type query: Tuple[bytes]
+        :return: response if found or None
+        :rtype: Tuple[bytes] | None
+        """
+        
+        if connections is None:
+            connections = self._connections
+
+        # Check every connection for query match
+        for connection in connections:
+            if (connections[connection].q).encode("utf-8") == query:
+
+                # function as written in yaml
+                func = connections[connection].function
+
+                for source in connections[connection].source_list:
+                    # Find source for connection
+                    if source["source_name"] == "self":
+                        value = self._properties[source["source_parameter"]]._value
+                        func = func.replace('%' + source["source_parameter"] + '%', str(value))
+                    else:
+
+                        for device in self.devices._internal:
+                            if device == source["source_name"]:
+                                # Find correct property
+                                for property in self.devices._internal[device]._properties:
+                                    if property == source["source_parameter"]:
+                                        # Replace parameter in func with value of property
+                                        value = self.devices._internal[device]._properties[property]._value
+                                        func = func.replace('%' + property + '%', str(value))
+                                        break
+                                break
```

Comment:
Rather than using `%` I would go with `{ }` like everywhere else. And you could then simply use string formatting.

Reply:
Yes, I will change this.